controller/users.go

```diff
@@ -77,6 +83,195 @@ func (c *UsersController) Show(ctx *app.ShowUsersContext) error {
 	})
 }
 
+// Create creates a user when requested using a service account token
+func (c *UsersController) Create(ctx *app.CreateUsersContext) error {
+
+	isSvcAccount := token.IsSpecificServiceAccount(ctx, []string{"online-registration"})
+	if !isSvcAccount {
+		log.Error(ctx, nil, "account used to call create api is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("account not authorized to create users."))
+	}
+	userUrl, err := c.createUserInKeycloak(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to create user in keycloak")
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+	log.Info(ctx, map[string]interface{}{
+		"user_url": userUrl,
+	}, "successfully created new user in keycloak")
+
+	// TODO: Handle error, check if there was actually a URL returned.
+	userURLComponents := strings.Split(*userUrl, "/")
+
+	// TODO: Check if we got a real UUID.
+	identityID, _ := uuid.FromString(userURLComponents[len(userURLComponents)-1])
+
+	identity, user, err := c.createUserInDB(ctx, identityID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to create user in DB")
+
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+
+	// finally, if all works, we create a user in WIT too.
+	witURL, err := c.config.GetWITURL(ctx.RequestData)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to create user in WIT")
+
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
+	}
+	identity.User = *user // being explicit
+	c.RemoteWITService.CreateWITUser(ctx.Context, ctx.RequestData, identity, witURL, identityID.String())
+
+	return ctx.OK(ConvertToAppUser(ctx.RequestData, user, identity))
+}
+
+func (c *UsersController) createUserInKeycloak(ctx *app.CreateUsersContext) (*string, error) {
+
+	// All the below attributs are mandatory:
+	// "username", "email", "cluster", "company", "fullName", "approved", "rhd_username")
+
+	userAttributes := ctx.Payload.Data.Attributes
+
+	keycloakUser := login.KeytcloakUserRequest{
```

Comment:
We also need to create a link to the user's RHD account in Keycloak. Otherwise the user won't able to login.
And let's also add the RHD username to User table.

Reply:
umm. but the RHD username would be same as the identity username , right?  Then why are we gonna persist that separately ?

Reply:
If we allow to change the username then it will be different than RHD name. We can address this in a separate issue/PR though.